azure/.terraform.lock.hcl

```diff
@@ -0,0 +1,65 @@
+# This file is maintained automatically by "terraform init".
+# Manual edits may be lost in future updates.
+
+provider "registry.terraform.io/hashicorp/azurerm" {
```

Comment:
do we want to check this in?

Reply:
When you execute the `terraform init` command it actually says:
```
Terraform has created a lock file .terraform.lock.hcl to record the provider
selections it made above. Include this file in your version control repository
so that Terraform can guarantee to make the same selections by default when
you run "terraform init" in the future.
```
The file doesn't contain sensitive data but it makes sure that the exact versions are selected every time. Maybe it's not doing much with the `versions.tf` file but can be considered as an extra step. What you think?